projects/nlp/clustering_runner.py

```diff
@@ -0,0 +1,363 @@
+#!/usr/bin/env python
+# ----------------------------------------------------------------------
+# Numenta Platform for Intelligent Computing (NuPIC)
+# Copyright (C) 2016, Numenta, Inc.  Unless you have purchased from
+# Numenta, Inc. a separate commercial license for this software code, the
+# following terms and conditions apply:
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero Public License version 3 as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+# See the GNU Affero Public License for more details.
+#
+# You should have received a copy of the GNU Affero Public License
+# along with this program.  If not, see http://www.gnu.org/licenses.
+#
+# http://numenta.org/licenses/
+# ----------------------------------------------------------------------
+
+import argparse
+import numpy
+import os
+import simplejson
+from textwrap import TextWrapper
+import matplotlib.pyplot as plt
+
+from htmresearch.support.csv_helper import readCSV, mapLabelRefs
+from htmresearch.frameworks.nlp.classification_model import ClassificationModel
+from htmresearch.frameworks.nlp.classify_document_fingerprint import (
+  ClassificationModelDocumentFingerprint
+)
+from htmresearch.frameworks.nlp.classify_fingerprint import (
+  ClassificationModelFingerprint
+)
+from htmresearch.algorithms.hierarchical_clustering import (
+  HierarchicalClustering
+)
+from htmresearch.frameworks.nlp.model_factory import (getNetworkConfig)
+
+SAVE_PATH = os.path.join(os.path.dirname(__file__),
+                         "results/HierarchicalClustering/")
+
+
+def runExperiment(args):
+  if not os.path.exists(SAVE_PATH):
+    os.makedirs(SAVE_PATH)
+  
+  (trainingDataDup, testData, labelRefs, documentCategoryMap,
+   documentTextMap) = readData(args)
+  
+  # remove duplicates from training data
+  includedDocIds = set()
+  trainingData = []
+  for record in trainingDataDup:
+    if record[2] not in includedDocIds:
+      includedDocIds.add(record[2])
+      trainingData.append(record)
+  
+  if args.modelName == "docfp":
```

Comment:
There's now a model factory in frameworks/nlp (see simple_labels for usage example).  Could you switch to that? It supports more model types.